pilot/cmd/pilot-agent/app/cmd.go

```diff
@@ -49,22 +50,9 @@ const (
 	localHostIPv6 = "[::1]"
 )
 
-// TODO: Move most of this to pkg options.
 var (
-	dnsDomain          string
-	stsPort            int
-	tokenManagerPlugin string
-
-	meshConfigFile string
-
-	// proxy config flags (named identically)
-	serviceCluster         string
-	proxyLogLevel          string
-	proxyComponentLogLevel string
-	concurrency            int
-	templateFile           string
-	loggingOptions         = log.DefaultOptions()
-	outlierLogPath         string
+	loggingOptions = log.DefaultOptions()
+	proxyArgs      *options.ProxyArgs
```

Comment:
why a pointer?

Reply:
> why a pointer?

Cause PilotArgs in the pilot-discover is also a pointer,I'm not sure why, so I do it here.
Would it be better to not use pointer here?